### PR TITLE
Add dockerized version of user mgmt service

### PR DESCRIPTION
**Pull request by shubhangis91**
Added code for dockerizing the service #9 #11

### CODE REVIEW

**Comment by adithyaselv** [comment]
I am getting an error 
```
ERROR: Service 'user-mgmt' failed to build: ADD failed: stat /var/lib/docker/tmp/docker-builder369896450/target/UserManagement-0.0.1-SNAPSHOT.jar: no such file or directory
``` 
when i do `docker-compose up -d`

am i doing something wrong ? 

`docker-compose build` also results in same error

Please fix readme to reflect on this change

**Comment by adithyaselv** [comment]
Also I think we must have tomcat running as a part of this container. So the behaviour after dockerising this sub-project will be the user does  `docker-compose up -d` and all the dependencies must get installed and the application must start inside the container.

File: UserManagement/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+FROM java:8
+FROM maven:alpine
+WORKDIR /app
+COPY . /app
+RUN mvn -v
```
Comment (adithyaselv) [inline UserManagement/Dockerfile]: do we need to this seeing version thing ?

File: UserManagement/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+RUN mvn clean install -DskipTests
+EXPOSE 8181
+ADD ./target/UserManagement-0.0.1-SNAPSHOT.jar /jar/
```
Comment (adithyaselv) [inline UserManagement/Dockerfile]: I think we can just create a maven goal to run the main class and then do something like `mvn exec:main` to run the application. also please check again if we need tomcat as a part of this image.

File: UserManagement/Dockerfile (hunk 1 above)
Comment (shubhangis91) [reply UserManagement/Dockerfile]: will remove that line

File: UserManagement/Dockerfile (hunk 2 above)
Comment (shubhangis91) [reply UserManagement/Dockerfile]: I think we wouldn't need the tomcat service for this image. we just need the jar that can be created with maven. looking into that right now.

**Comment by shubhangis91** [comment]
This is done. Updated the code.

**Comment by adithyaselv** [comment]
Did you also check if the kafka producer/consumer part of user management worked ? As far as what I understand it should not since kafka would be running a different network from this service.

**Comment by shubhangis91** [comment]
It was able to write to the topic mentioned in the configuration code. Does it need to be changed?

**Comment by adithyaselv** [comment]
Looks good

**Review by adithyaselv — APPROVED** [review]

**Pull request merged**